### PR TITLE
[Doc] Update the support of hostname verification in different clients

### LINKED ISSUE #14432: [Doc][Client] explanations for hostname verification configuration are inaccurate
Here is an issue from Pulsar users:

🔷🔷🔷

Please consider this page, https://pulsar.apache.org/docs/en/security-tls-transport/#cli-tools

```
By default, Pulsar clients disable hostname verification, as it requires that each broker has a DNS record and a unique cert.

The examples below show hostname verification being disabled for the Java client, though you can omit this as the client disables the hostname verification by default. C++/python/Node.js clients do now allow configuring this at the moment.
(See the screenshot for the examples)
```

I am trying to understand why the CLI and Java sample configuration disable hostname verification, but the Python and C++ sample configurations enable hostname verification.

If as quoted, “By default, Pulsar clients disable hostname verification”, and “C++/python/Node.js clients do now allow configuring this at the moment.“, then the Python and C++ sample configurations do nothing when they attempt to set hostname verification to true?

If that is true, why have such confusing sample configurations?

Normally, sample configurations are identical across different programming languages. Why in this case is “enableTlsHostnameVerification” set to different values across different programming languages?

🔷🔷🔷

Feedback from @BewareMyPower: The default value in C++ client is false. I'm also confused by these examples and the document. I think it's more like an issue about documents.

### PR DESCRIPTION
**Pull request by momo-jun**
### Modifications
Fix #14432.

1. Streamline the default value of configuration to false.
2. Update the code sample for the Node.js client.
3. Add a note for the C# client.

Note that the changes will be applied to more historical doc versions since 2.7.0. 

### Documentation
- [x] `doc`

### CODE REVIEW

**Comment by momo-jun** [comment]
Hi @BewareMyPower @RobertIndie @tuteng @Anonymitaet can you please review this doc PR? Thanks.
Please kindly note that the changes will be applied to more historical doc versions since 2.7.0 after the PR gets approval.

**Review by BewareMyPower — APPROVED** [review]

File: site2/docs/security-tls-transport.md [hunk 1]
```diff
@@ -256,3 +259,4 @@ var client = PulsarClient.Builder()
                          .VerifyCertificateName(false)     //Default is 'false'
                          .Build();
 ```
+> Note that `VerifyCertificateName` refers to the configuration of hostname verification in the C# client.
```
Comment (RobertIndie) [inline site2/docs/security-tls-transport.md:262]: @blankensteiner Hi, could you help verify this?

**Review by RobertIndie — APPROVED** [review]

File: site2/docs/security-tls-transport.md (hunk 1 above)
Comment (blankensteiner) [reply site2/docs/security-tls-transport.md:262]: Hi @RobertIndie 
I can verify that by default it does not verify the certificate name.

Comment (RobertIndie) [reply site2/docs/security-tls-transport.md:262]: @blankensteiner Is `VerifyCertificateName` the same as the hostname verification in other language clients? It looks like they are all checking the `CommonName` in the certificate.

Comment (blankensteiner) [reply site2/docs/security-tls-transport.md:262]: @RobertIndie I have no idea. We register a [callback](https://docs.microsoft.com/en-us/dotnet/api/system.net.security.remotecertificatevalidationcallback?view=net-6.0) when creating the [SslStream](https://docs.microsoft.com/en-us/dotnet/api/system.net.security.sslstream.-ctor?view=net-6.0#system-net-security-sslstream-ctor(system-io-stream-system-boolean-system-net-security-remotecertificatevalidationcallback-system-net-security-localcertificateselectioncallback)) and then ignore the RemoteCertificateNameMismatch flag on the [SslPolicyErrors](https://docs.microsoft.com/en-us/dotnet/api/system.net.security.sslpolicyerrors?view=net-6.0) if VerifyCertificateName is set to false.

**Pull request merged**